ow_power_system/src/power_system_node.cpp

```diff
@@ -124,26 +125,23 @@ int main(int argc, char* argv[]) {
         std::sort(samplesSOC.begin(), samplesSOC.end());
         double soc_median = samplesSOC.at(samplesSOC.size() / 2);
         soc_msg.data = soc_median;
-      }
 
-      // Temperature Code
-      //std::vector<UData> systemStates = eod_event.getSystemState()[0]; // Get the system (battery) state at t=0 (now)
-      // This is in format [stateId][sampleId].
-      // Here you will have to choose the sampleId the corresponds to your chosen percentile (e.g., Median) - and load it into a vector<double> where each element is a different state (in same order as above). 
-      // For Example:
-      //std::vector<double> medianSystemState(systemStates.size());
-      //for (int i = 0; i < medianSystemState.size(); i++) 
-        //medianSystemState[i] = systemStates[i][MEDIAN]; // Where MEDIAN corresponds to the sampleID of the median EOL prediction
-        
-      // Next, You will pass it into the model outputEqn to get the outputs:
-      //auto output = model.outputEqn(medianSystemState);
-      //double temperature = output[1];
-      //tempbat_msg.data = temperature;
+        // Temperature Code
+        auto stateSamples = eod_event.getSystemState()[0];
+        std::vector<double> state;
+        for (auto sample : stateSamples) {
+          state.push_back(sample[0]);
+        }
+        auto& model = ((ModelBasedPrognoser *) prognoser.get())->getModel();
```

Comment:
Better to use dynamic_cast here
```suggestion
        const auto& model = dynamic_cast<ModelBasedPrognoser*>(prognoser.get())->getModel();
```